pkg/haprovider/haprovider.go

```diff
@@ -165,7 +180,19 @@ func (r *HAProvider) getAviInfraSettingFromCluster(ctx context.Context, cluster
 		return nil, nil
 	}
 
-	aviInfraSettingName := GetAviInfraSettingName(&adcForCluster[0])
+	return &adcForCluster[0], nil
```

Comment:
What if `adcForCluster` is nil?

Reply:
You should handle those conditons

Reply:
The value is returned from this [function](https://github.com/vmware-samples/load-balancer-operator-for-kubernetes/blob/87458b75b21a6bf6f83848a38e91b3e8311efd9b/pkg/controller-runtime/handlers/cluster_for_akodeploymentconfig_handler.go#L61),  it the returned value is list type, so is there still a need for checking the nil?

Reply:
How about this: https://go.dev/play/p/33esyHiugnK

Reply:
The line above already handled the case where the list is empty or nil. I tried the same on the playground, https://go.dev/play/p/GF7AMlmy8_U

Reply:
Good! Above line compressed by github review board